browser/components/newtab/RemoteAboutNewTab.jsm

```diff
@@ -331,6 +343,14 @@ let RemoteAboutNewTab = {
     }
   },
 
+  setEnabled: function(name, data) {
+    this.pageListener.sendAsyncMessage("NewTab:setEnabled", data.enabled);
```

Comment:
following the suggested changes above, in NewTabPrefsProvider.jsm, you wouldn't need to unpack any object